### PR TITLE
Test refactor and spies

### LINKED ISSUE #23: Fix User-Test File
Now that we have converted form using names to ID's we need to make sure our tests still pass/are felxible enough.

### PR DESCRIPTION
**Pull request by timkeresey**
###### What does this PR do?  
- Create spies for dom manipulation methods.
- Refactor class file tests

###### Where should the reviewer start?  
- Line 1 of spies file.

###### How should this be manually tested?  
- By running npm test in the terminal

###### Any background context you want to provide?  
- The spy tests require that the imports in the domUpdates file be commented out for them to run without error. They are currently commented out and will need to be undone for the app to run.

###### What are the relevant tickets?  
- closes #23  and #33

### CODE REVIEW

File: src/domUpdates.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
-import scripts from './scripts';
-import Pantry from './pantry';
-import Recipe from './recipe';
-import User from './user';
-import Cookbook from './cookbook';
+// import scripts from './scripts';
```
Comment (nathanielmillard) [inline src/domUpdates.js:1]: Now that we have time we may check in with educators to see if there is a work around for this?

File: test/pantry-test.js [hunk 2]
```diff
@@ -24,17 +24,11 @@ describe('Pantry', () => {
...
-    expect(pantry.userIngredients.length).to.equal(users[0].pantry.length);
+    expect(pantry.userPantry.length).to.equal(users[0].pantry.length);
   });
 
-  it.only('Should update the user pantry when a recipe has been cooked', () => {
-    pantry.updatePantry(recipe1);
+  it('should convert array into ingredients object', () => {
     pantry.createPantry(ingredientsData);
-    expect(pantry.userPantry[1].amount).to.equal(0);
+    expect(pantry.userIngredients).to.be.an('object');
```
Comment (nathanielmillard) [inline test/pantry-test.js:32]: Glad to see these are updated since we changed this functionality a few times.

**Review by nathanielmillard — APPROVED** [review]
Over all looking good, and glad to see these tests up to date again.

**Review by nathanielmillard — APPROVED** [review]
Glad to see we got that one line resolved.

**Review by stacyp2006 — APPROVED** [review]
Glad we are passing all the tests. Nice work!

**Pull request merged**